src/analysis/AnalysisClient.js

```diff
@@ -93,21 +95,71 @@ class AnalysisClient {
     analyzed_items_file.end();
   }
 
+
+  assignRepresentativeWeightToItem(pairedProductsList){
+    /*
+    If we have seen at least five item in a category, assign 3/4 of the max weight in a category to an item 
+    in that category whose weight is unknown
+    The choice of 3/4 of max is arbitrary and can be changed in the future.    
+    If we have seen less than 5, use the max weight instead because we don't know enough about weights in category
+    */
+    let representativeCount = 5;
+    let representativeWeight = 0.75;
+    let categoriesRepWeight  = {};
+    let categoriesCount  = {};
+    
+    pairedProductsList.products.forEach(function(pairedProduct){
+      //if weight is known and categoryId is defined
+      if (pairedProduct.amazonProd.dimensions.weight != 'UNKNOWN' && pairedProduct.amazonProd.bestSalesRanking.categoryId) {
```

Comment:
I found the category Ids to be garbage and ended up [using an attribute called the Product Group to determine sales categories](https://github.com/sulaimonlasisi/clearance-app/blob/master/src/amazon/AmazonProduct.js#L17) so you should use that here as well.
Also, I don't think the sales ranking category check is even necessary since data with an Unknown category is filtered out earlier by the [`isPopular` method](https://github.com/sulaimonlasisi/clearance-app/blob/master/src/amazon/AmazonProduct.js#L34). I'd test that to be sure though.